### ISSUE TITLE
Known defect: HRV Status Not Entering Ready Status - Multiple models

### ISSUE THREAD

**Issue by spvandermark** [dliedke/Meditate#13]
When selecting a meditation session, HRV stuck in “Waiting HRV” even with after holding light button for 15s then turning on watch again and retrying the app.  Worked fine until the latest Garmin update. 

Forerunner 965
Software version 20.26

**Comment by dliedke** [comment]
Yes, same issue here, I already did a lot of fixes but the issue always come back

**Comment by spvandermark** [comment]
Tried your test and it do not work the first few times. Tried minutes  later and it worked. Very random.  

I think it's issues with the software update. Researched and there seems a lot of introduced issues with HR and one specific for HRV. 

https://forums.garmin.com/sports-fitness/running-multisport/f/forerunner-265-series/385137/after-latest-software-hrv-and-sleep-tracking-have-gaps

**Comment by spvandermark** [comment]
The shutdown and restart is also randomly working.

**Comment by dliedke** [comment]
From:
https://forums.garmin.com/sports-fitness/running-multisport/f/forerunner-965/377764/hrv-gaps-the-solution

What is clear:

- It is not a hardware issue: lot of people have watches replaced (including me) and it is still happening.

- If you restart, you have clean readings for a couple of weeks, then back to the same.

- It is affecting several models (epix, fenix, forerunner)

- Cannot be attributed to sleeping patterns, since this wasn't happening for most before.

HRV is based on beat-to-beat metric directly from the watch. If watch is not providing it, the Meditate will not detect HRV.

It is a pain indeed, used to be better. Now even respiration rate is no longer working as well, even when HRV is detected.

**Comment by spvandermark** [comment]
20.29 was pushed out today and I was hoping it was addressed. Not so much still same issues.

**Comment by dliedke** [comment]
Hello, I tried a lot of different approaches to fix this and no luck. I will keep this issue open forever. 

Sorry about that

**Comment by spvandermark** [comment]
I'll keep on the lookout for any updates you try. Feel free to reach out if you need something tested. Keep up the good work.

**Comment by dliedke** [comment]
Thanks, so far for me the only way to fix the issue with HRV detection is really soft reset the watch. But even so sometimes it doesn't work.
